### ISSUE TITLE
[FR] `legend = false` should not hide colorbar

### ISSUE THREAD

**Issue by mkborregaard** [JuliaPlots/Plots.jl#2756]
Right now `legend = false` will also hide the colorbar, leading to the awkward `legend = false, colorbar = true`.  I don't think we ever want the current behaviour. legend and colorbar should simply be separate IMHO

**Comment by isentropic** [comment]
At this point it has the default value `colobar = :legend` that copies the behavior of the legend. As per https://github.com/JuliaPlots/Plots.jl/blob/fff8f84acf62550f068810970e146e085264c9c5/src/arg_desc.jl#L107

I guess we should set the default value to be `colorbar = :right` should it be the case though?

**Comment by isentropic** [comment]
Personally, I think the current behavior is very much reasonable, I wonder what others think

**Comment by mkborregaard** [comment]
The colorbar already has sensible defaults - it's being shown if you ask for it explicitly, if you have a heatmap or a surface, or if you have color encoding in a variable with `line_z` or `marker_z`. I don't see why `legend` should override those sensible defaults, to be honest.

**Comment by mkborregaard** [comment]
```julia
plot(rand(4,5), line_z = (1:5)')
```
![Skærmbillede 2020-06-04 kl  09 08 44](https://user-images.githubusercontent.com/8429802/83726051-40e52480-a643-11ea-8dd1-4e0ca27c2100.png)

```julia
plot(rand(4,5), line_z = (1:5)', legend = false)
```
![Skærmbillede 2020-06-04 kl  09 08 54](https://user-images.githubusercontent.com/8429802/83726070-480c3280-a643-11ea-84b6-1b8d67e5dc09.png)

BTW I think maybe 
```
┌ Warning: Multiple series with different color share a colorbar. Colorbar may not reflect all series correctly.
└ @ Plots ~/.julia/packages/Plots/JKY3H/src/backends/gr.jl:491
```
should have been an `@info` not a warning?

**Comment by isentropic** [comment]
Good point, perhaps, we would want no legend if colorbar needs to be drawn?
That is, the default behaviour should be no legend in the presence of colorbar?

**Comment by daschw** [comment]
> Additionally, would we want no legend if colorbar needs to be drawn?
That is, the default behaviour should be no legend in the presence of colorbar?

No, I don't think so. For heatmaps and contours legends are already suppressed automatically and for other cases (see Michael's first plot) it could make sense to have a legend. In that case I would leave removing the legend to the users.

**Comment by mkborregaard** [comment]
Agree with Daniel. The issue is also there with composite plots 
![amphibians](https://user-images.githubusercontent.com/8429802/83742142-57967600-a659-11ea-99ba-ecb7bbe9fbdc.jpg)
(e.g. if I want to combine a heatmap with a polygon such as in the below example), but that should also be solved by unlinking them.

**Comment by daschw** [comment]
Nice plot! What does it show?

**Comment by mkborregaard** [comment]
Species richness of amphibians :-)
